### PR TITLE
Update cert-manager to use workload identities for authentication

### PR DESCRIPTION
**Pull request by phillebaba**
This adds a new user assigned identity as it was using the same resources as external-dns before. This change also updates the cert-manager version as the older version does not support the new auth method.

### CODE REVIEW

File: modules/azure/aks/variables.tf [hunk 1]
```diff
@@ -204,3 +204,13 @@ variable "log_eventhub_authorization_rule_id" {
   description = "The authoritzation rule id for event hub"
   type        = string
 }
+
+variable "dns_zones" {
```
Comment (CalleB3) [inline modules/azure/aks/variables.tf:208]: might wanna call this `dns_zone` to keep the same naming in every module?

Comment (phillebaba) [reply modules/azure/aks/variables.tf:208]: The name for all other variables is wrong. Was planning on updating the other modules. This is a list of zones not a singe zone.

Comment (CalleB3) [reply modules/azure/aks/variables.tf:208]: True, that makes more sense

**Review by CalleB3 — APPROVED** [review]
LGTM

**Pull request merged**